homeassistant/components/remote/roku.py

```diff
@@ -0,0 +1,64 @@
+"""
+Support for the Roku remote.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/remote.roku/
+"""
+
+from homeassistant.components import remote
+from homeassistant.const import (CONF_HOST)
+
+
+DEPENDENCIES = ['roku']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Roku remote platform."""
+    if not discovery_info:
+        return
+
+    host = discovery_info[CONF_HOST]
+    async_add_entities([RokuRemote(host)])
+
+
+class RokuRemote(remote.RemoteDevice):
+    """Device that sends commands to an Roku."""
+
+    def __init__(self, host):
+        """Initialize the Roku device."""
+        from roku import Roku
+
+        self._roku = Roku(host)
+        self._unique_id = self._roku.device_info.sernum
```

Comment:
This is doing I/O, so either change to sync api in setup, ie `setup_platform` or move this call to `update` and call `async_add_entities(entities, True)`. I suggest the latter to keep consistency.